// Copyright 2022 The Bucketeer Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package bucketeer.eventcounter;
option go_package = "github.com/bucketeer-io/bucketeer/proto/eventcounter";

import "proto/eventcounter/histogram.proto";

message DistributionSummary {
  double mean = 1;
  double sd = 2;
  double rhat = 3;
  Histogram histogram = 4;
  double median = 5;
  double percentile025 = 6;  // Remove "_" because python should generate same
                             // field name for Datasource.
  double percentile975 = 7;  // Remove "_" because python should generate same
                             // field name for Datasource.
}
